debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/views/memory/renderings/RenderingsUtil.java

```diff
@@ -558,21 +558,7 @@ public static byte[] convertHexStringToByteArray(String str, int numBytes, int n
 			String oneByte = str.substring(i*2, i*2+2);
 
 			Integer number = Integer.valueOf(oneByte, 16);
-			if (number.compareTo(Integer.valueOf(Byte.toString(Byte.MAX_VALUE))) > 0)
-			{
-				int temp = number.intValue();
-				temp = temp - 256;
-
-				String tempStr = Integer.toString(temp);
-
-				Byte myByte = Byte.valueOf(tempStr);
-				bytes[i] = myByte.byteValue();
-			}
-			else
-			{
-				Byte myByte = Byte.valueOf(oneByte, 16);
-				bytes[i] = myByte.byteValue();
-			}
+			bytes[i] = number.byteValue();
```

Comment:
Yes. Seeing this now, I realize we could also omit the boxing into "number". `bytes[i] = (byte) Integer.parseInt(oneByte, 16);`.

Even the `oneByte` substring is not really necessary (since Java 9 there is parseInt(CharSequence s, int beginIndex, int endIndex, int radix)), but personally I'd leave that for clarity.

Reply:
I have at least incorporated your proposal to directly cast to byte, as that's safe to do here, slightly improves performance, and, in my opinion, is still comprehensible: bd09a00

Reply:
LGTM.

Reply:
Thanks, Heiko!